src/components/FindMovie/FindMovie.tsx

```
@@ -1,10 +1,81 @@
-import React from 'react';
+import { ChangeEvent, FC, FormEvent, useState } from 'react';
+import cn from 'classnames';
 import './FindMovie.scss';
+import { Movie } from '../../types/Movie';
+import { getMovie } from '../../api';
+import { MovieCard } from '../MovieCard';
+import { MovieData } from '../../types/MovieData';
+
+interface Props {
+  onAddMovie: (movie: Movie) => void;
+}
+
+export const FindMovie: FC<Props> = ({ onAddMovie }) => {
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [query, setQuery] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const message = "Can't find a movie with such a title";
+
+  const findMovie = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setIsLoading(true);
+    getMovie(query)
+      .then(data => {
+        if (data.Response === 'False') {
+          setErrorMessage(message);
+          setMovie(null);
+
+          return;
+        }
+
+        const {
+          Title: title,
+          Plot: description,
+          Poster,
+          imdbID: imdbId,
+        } = data as MovieData;
+
+        const imgUrl =
+          !Poster || Poster === 'N/A'
+            ? 'https://via.placeholder.com/360x270.png?text=no%20preview'
+            : Poster;
+
+        const newMovie = {
+          title,
+          description,
+          imgUrl,
+          imdbUrl: `https://www.imdb.com/title/${imdbId}`,
+          imdbId,
+        };
+
+        setMovie(newMovie);
```

After a successful fetch you set the `movie` preview but you don't clear any previous `errorMessage`. While the preview hides the error visually, it's safer to explicitly clear the error when the fetch succeeds: call `setErrorMessage('')` after creating/setting `newMovie` to avoid stale error state if UI or tests check the error state programmatically.